core/src/main/java/org/wso2/testgrid/core/command/RunTestPlanCommand.java

```diff
@@ -144,18 +149,26 @@ public void execute() throws CommandExecutionException {
             TestPlan testPlan = generateTestPlan(deploymentPattern, testConfig, scenarioRepoDir, infraRepo);
 
             //Set the log file path
-            LogFilePathLookup.setLogFilePath(deriveLogFilePath(product, testPlan));
+            LogFilePathLookup.setLogFilePath(deriveLogFilePath(testPlan));
 
             // Product, deployment pattern, test plan and test scenarios should be persisted
             testPlan = persistTestPlan(testPlan);
 
             // Execute test plan
-            executeTestPlan(testPlan, infrastructure);
+            testPlan = executeTestPlan(testPlan, infrastructure);
+
+            // Generate report for the test plan
+            TestReportEngine testReportEngine = new TestReportEngine();
+            testReportEngine.generateReport(testPlan);
         } catch (IllegalArgumentException e) {
             throw new CommandExecutionException(StringUtil.concatStrings("Channel ",
                     channel, " is not defined in the available channels enum."), e);
         } catch (IOException e) {
             throw new CommandExecutionException("Error in reading file generated config file", e);
+        } catch (ReportingException e) {
+            throw new CommandExecutionException("Error in generating report for the test plan.", e);
+        } catch (TestGridException e) {
+            throw new CommandExecutionException("Error in deriving log file path.", e);
```

Comment:
We cannot say TestGridException will be thrown only for "Error in deriving log file path"! How about we throw a specific exception from LogFilePathLookup.setLogFilePath and catch that here?

Possible signature for the exception class: `public class LogFileDeriveException extends TestGridException`

Reply:
In this case, `TestGridException` will be only thrown for this reason. Furthermore, the error actually occurs when creating a UUID using infra params (not due to actually setting the log file path). Hence it is TestGrid core related than logging framework related.

Reply:
What will when this code evolves? What if someone adds a new logic that throws the TestGridException? I do not think he will notice to update the log.

Reply:
Added a custom exception `TestGridLoggingException`